### PR TITLE
Change WriteSeries signature

### PR DESCRIPTION
**Pull request by corylanou**
Change WriteSeries signature from:

```go
WriteSeries(database, retentionPolicy, name string, tags map[string]string, timestamp time.Time, values map[string]interface{}) (uint64, error)
```
To
```go
WriteSeries(database, retentionPolicy string, points ...influxdb.Point) (uint64, error)
```

### CODE REVIEW

File: collectd/collectd_test.go [hunk 1]
```diff
@@ -14,22 +15,18 @@ import (
...
-	values                          map[string]interface{}
+	database        string
+	retentionPolicy string
+	points          []influxdb.Point
 }
 
 var responses = make(chan *serverResponse, 1024)
 
-func (testServer) WriteSeries(database, retentionPolicy, name string, tags map[string]string, timestamp time.Time, values map[string]interface{}) (uint64, error) {
+func (testServer) WriteSeries(database, retentionPolicy string, points ...influxdb.Point) (uint64, error) {
```
Comment (benbjohnson) [inline collectd/collectd_test.go]: Why did you decide to use varargs? It seems like if we'd pass a bunch of points in then they'd likely already be in a slice.

Comment (otoolep) [inline collectd/collectd_test.go]: Ditto. I was expecting a slice too, when I heard about this work. Nothing stops us passing a single point as a single-entry slice.

Comment (corylanou) [inline collectd/collectd_test.go]: If the majority of the time we'll always have many, then we can make it a slice.  It felt like we would be doing single point writes a lot as well from other endpoints (graphite for example).  Using vargs allows you to send in single or multiple.

Not married to it, if you have a preference, I can go either way.

**Comment by otoolep** [comment]
Looks sane, but I also prefer an API that expects a slice.

File: server.go [hunk 2]
```diff
@@ -1210,9 +1211,22 @@ type createSeriesIfNotExistsCommand struct {
 	Tags     map[string]string `json:"tags"`
 }
 
+type Point struct {
```
Comment (benbjohnson) [inline server.go]: Add a godoc line.

**Comment by otoolep** [comment]
I didn't realise you could still send in a slice. Interesting. I'll leave it to @benbjohnson and @corylanou to decide, seems more like a style question now.

File: server.go [hunk 3]
```diff
@@ -1210,9 +1211,22 @@ type createSeriesIfNotExistsCommand struct {
...
+	Name      string
+	Tags      map[string]string
+	Timestamp time.Time
+	Values    map[string]interface{}
+}
+
 // WriteSeries writes series data to the database.
 // Returns the messaging index the data was written to.
-func (s *Server) WriteSeries(database, retentionPolicy, name string, tags map[string]string, timestamp time.Time, values map[string]interface{}) (uint64, error) {
+func (s *Server) WriteSeries(database, retentionPolicy string, points ...Point) (uint64, error) {
```
Comment (benbjohnson) [inline server.go]: I don't have a strong preference toward `[]*Point` vs `[]Point` until we benchmark. It's less GC load if it's not pointers. It's much faster to sort if it is pointers. We'll just have to benchmark it later and see. No need to change anything right now.

**Comment by benbjohnson** [comment]
:+1: for slice instead of varargs.

**Comment by benbjohnson** [comment]
a few minor nits but lgtm.

**Pull request merged**